pages/post/[postId].tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect, useState } from 'react';
+import { useRouter } from 'next/router';
+import { getShortUrl, getPostDate } from '../../utils/utils';
+
+interface PostProps {
+  postID: string;
+  author: {
+    userName: string;
+  };
+  title: string;
+  url: URL;
+  description: string;
+  upvotes: number;
+  timeCreated: number;
+}
+
+const ViewPost = () => {
+  const router = useRouter();
+  const { postId } = router.query;
+  const [post, setPost] = useState<PostProps>();
+
+  useEffect(() => {
+    const getPost = async () => {
+      try {
+        const response = await fetch('/api/getAllPosts');
+        const allData = await response.json();
+        const { posts } = allData.data;
+        setPost(
+          () => posts.filter((post: PostProps) => post.postID === postId)[0]
```

Comment:
Use find `() => posts.find((post: PostProps) => post.postID === postId)`. It'll return the first object that matches the criteria